engine/apps/grafana_plugin/apps.py

```diff
@@ -0,0 +1,33 @@
+import logging
+import sys
+
+from django.apps import AppConfig, apps
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class GrafanaPluginConfig(AppConfig):
+    name = "apps.grafana_plugin"
+
+    def ready(self):
+        """
+        For OSS installations, validate that OSS_GRAFANA_API_URL environment variable is specified, otherwise
+        abort app startup.
+
+        We only care to run this for OSS_INSTALLATIONS. The `"runserver" in sys.argv` check is to avoid running this
+        for the django migrate command. For a fresh installation this would crash because user_management table would
+        [not exist](https://stackoverflow.com/a/63326719).
+        """
+        if "runserver" in sys.argv and settings.OSS_INSTALLATION is True:
+            Organization = apps.get_model("user_management", "Organization")
+            has_existing_org = Organization.objects.first() is not None
+
+            # only enforce the following for new setups - if no organization exists in the database
+            # and the OSS_GRAFANA_API_URL env var is not specified, exit the application
+            if has_existing_org is False and settings.SELF_HOSTED_SETTINGS["GRAFANA_API_URL"] is None:
+                logger.error(
+                    f"For OSS installations, OSS_GRAFANA_API_URL is a required environment variable. Please set it and restart the application."
+                )
+                sys.exit()
```

Comment:
I took a bit of an opinionated approach to this but I think we should crash the app on OSS startup if certain conditions are not met, like this one. ([`AppConfig.ready` docs](https://docs.djangoproject.com/en/4.1/ref/applications/#django.apps.AppConfig.ready))